.github/actions/publish-nuget/Add-SourceLinkPackage.ps1

```diff
@@ -17,5 +17,32 @@ else
 
 foreach ($projectFile in $projectFiles)
 {
-    dotnet add $projectFile.FullName package 'Microsoft.SourceLink.GitHub' --source 'https://api.nuget.org/v3/index.json'
+    # Below we first prepare the project file by adding the NuGetBuild=true property to the top of it. This is needed
+    # for dotnet add package which could otherwise fail due to conditions in the project file.
+
+    # Load the project file as XML.
+    $projectXml = [xml](Get-Content $projectFile)
+
+    # Define the xmlns to access the elements in the csproj.
+    $ns = New-Object Xml.XmlNamespaceManager $projectXml.NameTable
+    $ns.AddNamespace('ns', $projectXml.DocumentElement.NamespaceURI)
+
+    # Create a new NuGetBuild property in a new PropertyGroup.
+    $propertyGroup = $projectXml.CreateElement('PropertyGroup', $projectXml.DocumentElement.NamespaceURI)
+    $nuGetBuildNode = $projectXml.CreateElement('NuGetBuild', $projectXml.DocumentElement.NamespaceURI)
+    $nuGetBuildNode.InnerText = 'true'
+    $propertyGroup.AppendChild($nuGetBuildNode)
+
+    # Add the new PropertyGroup to the project as the first node, to make sure that it's fully applied.
+    $projectXml.Project.InsertBefore($propertyGroup, $projectXml.Project.FirstChild)
+
+    # Save the changes back to the .csproj file.
+    $projectXml.Save($projectFile)
+
+    # --no-restore because we run it in a separate step with a configured output verbosity, what's not possible with
+    # the package command (and letting it implicitly restore here also makes NuGet publishing take 2-3 times as much).
+    # Unfortunately, this also makes it skip compatibility checks.
+    dotnet add $projectFile.FullName package 'Microsoft.SourceLink.GitHub' --source 'https://api.nuget.org/v3/index.json' --no-restore
```

Comment:
I see, that's a pity.

With implicit restore, this command will do a package restore with a very verbose output (it looks more verbose than the default `normal` for `dotnet` commands, and definitely a lot more verbose than `quiet`, what's the default for our workflows). Most possibly it's partially this, that it slows NuGet package publishing down a lot (not just this step, but the whole thing will take 2-2,5 minutes instead of <1 minute for UITT, for example).

Reply:
It's more like 2000 extra lines, see e.g. [here](https://github.com/Lombiq/UI-Testing-Toolbox/actions/runs/8884157839/job/24436095344#step:4:112) (contrast this with e.g. [this other one](https://github.com/Lombiq/UI-Testing-Toolbox/actions/runs/8898351256/job/24435314424#step:4:110)). Discarding the whole output is not an option, because we need to know about that information in case something fails. I really don't want to go down the "filter the stdoutput for anything that's always useful" route, and it wouldn't help with the performance anyway.

But this is just FYI, it doesn't matter if we need to use it to work with OCC.

Reply:
I will have to look into this, but I vaguely remember that slow console out is a Windowsispecific issue while the publish-nuget runs on Ubuntu. Anyway, as you said not important for now.